packages/block-library/src/page-list/edit.js

```diff
@@ -52,6 +51,34 @@ export default function PageListEdit( { context, clientId } ) {
 		style: { ...context.style?.color },
 	} );
 
+	const makeBlockTemplate = ( parentId = 0 ) => {
+		const pages = pagesByParentId.get( parentId );
+
+		if ( ! pages?.length ) {
+			return [];
+		}
+
+		return pages.reduce( ( template, page ) => {
+			const hasChildren = pagesByParentId.has( page.id );
+			const pageProps = {
+				id: page.id,
+				label: page.title?.rendered,
+				title: page.title?.rendered,
+				link: page.url,
+				hasChildren,
+			};
+			let item = null;
+			const children = makeBlockTemplate( page.id );
+			item = [ 'core/page-list-item', pageProps, children ];
+
+			template.push( item );
+
+			return template;
+		}, [] );
+	};
+
+	const pagesTemplate = makeBlockTemplate();
```

Comment:
It would be a good idea to memoize the inner block template; otherwise, `useInnerBlockTemplateSync` will try to synchronize it on every rerender.

Reply:
Yes! 💯 👍🏻